src/GraphQL/Federation/Extensions/FederationMetadataExtensions.cs

```diff
@@ -0,0 +1,125 @@
+using GraphQL.Types;
+using static GraphQL.Federation.FederationHelper;
+
+namespace GraphQL.Federation;
+
+/// <summary>
+/// Extension methods to configure federation directives within GraphQL schemas.
+/// These methods allow the application of various federation directives such as @key, @shareable, @inaccessible, @override, @external, @provides, and @requires to GraphQL types and fields.
+/// </summary>
+public static class FederationMetadataExtensions
+{
+    /// <summary>
+    /// Adds the "@key" directive to a GraphQL type.
+    /// </summary>
+    /// <param name="graphType">The GraphQL type to which the directive is added.</param>
+    /// <param name="fields">An array of field names that form the key.</param>
+    /// <param name="resolvable">Indicates whether the key is resolvable. Default is true.</param>
+    /// <typeparam name="TMetadataWriter">The type of the metadata writer.</typeparam>
+    /// <returns>The modified metadata writer.</returns>
+    public static TMetadataWriter Key<TMetadataWriter>(this TMetadataWriter graphType, string[] fields, bool resolvable = true)
```

Comment:
The use of extension methods in constructor looks somewhat ugly because it requires using `this.`. Are the any other alternatives?

Reply:
> The use of extension methods in constructor looks somewhat ugly because it requires using `this.`. Are the any other alternatives?

The only other option is to add methods directly on the class.  This requires duplicating all of the methods that apply to graph types within `ComplexGraphType`.

Personally I'd like to treat federation as separate, as if it was in a separate NuGet package, which may be the case in the future.  In that vein, everything is in the GraphQL.Federation namespace and should be almost completely separable from the main GraphQL.NET codebase, similar to the design of the data loaders.

Reply:
> I think it may be useful to add a link to the directive explanation in the XML docs https://www.apollographql.com/docs/federation/federated-types/federated-directives

done